src/autoreplace_gui.cpp

```diff
@@ -175,6 +176,12 @@ class ReplaceVehicleWindow : public Window {
 				if (!CheckAutoreplaceValidity(this->sel_engine[0], eid, _local_company)) continue;
 			}
 
+
+			if (!draw_left && e->IsHidden(_local_company)) {
```

Comment:
Can't go in the else above ?

Reply:
I generally prefer having unrelated checks separate. Though in this case they're already mixed up.